MIGRATION.md

```diff
@@ -0,0 +1,278 @@
+# Migration from 1.x to 2.0
+
+This document describes the main changes introduced in SDK `2.0` compared to `1.x`.
+
+### Product Modules 
+
+All the products (RUM, Trace, Logs, etc.) are now extracted into different modules. That allows you to integrate only what is needed into your application.
```

Comment:
we should include package managers code snippets here somewhere. Places like https://docs.datadoghq.com/logs/log_collection/ios/?tab=cocoapods will not be updated until we reach GA and to make migration smoother (copy-paste basically), we should add snippets each package.